dask_kubernetes/operator/controller/controller.py

```diff
@@ -597,17 +606,51 @@ async def daskworkergroup_replica_update(
             )
         logger.info(f"Scaled worker group {name} up to {desired_workers} workers.")
         if workers_needed < 0:
-            worker_ids = await retire_workers(
-                n_workers=-workers_needed,
-                scheduler_service_name=f"{cluster_name}-scheduler",
-                worker_group_name=name,
-                namespace=namespace,
-                logger=logger,
-            )
-            logger.info(f"Workers to close: {worker_ids}")
-            for wid in worker_ids:
-                worker_deployment = await Deployment(wid, namespace=namespace)
-                await worker_deployment.delete()
+            len_current_workers = len(current_workers)
+            deleted_workers = []
+            for idx in range(-workers_needed):
+                if idx > len_current_workers:
+                    break
+                worker = current_workers[idx]
+                kube_pods = await worker.pods()
+                assert (
+                    len(kube_pods) <= 1
+                ), "Deployment cannot have more than 1 replica pod"
+
+                if not kube_pods:
+                    continue
+
+                worker_pod = kube_pods[0]
+                if (
+                    worker_pod.status.phase
+                    not in [
+                        "ContainerCreating",
+                        "Initializing",
+                        "Terminating",
+                        "Running",
```

Comment:
I think you might be confusing the Pod phases with the status that `kubectl` displays. The possible values for phase are `Pending`, `Running`, `Succeeded`, `Failed` and `Unknown`.

https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.28/#podstatus-v1-core

Reply:
Yup, you're correct I've been debugging this using k9s